File: 14-longest-common-prefix/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	str := strs[0]
	for i := 1; i < len(strs); i++ {
		Max := 0
		maxLen := len(str)
		if len(strs[i]) < maxLen {
			maxLen = len(strs[i])
		}
		for j := 0; j < maxLen; j++ {
			if strs[i][j] == str[j] {
				Max++
			} else {
				break
			}
		}
		if Max == 0 {
			return ""
		}
		str = str[0:Max]
	}
	return str
}

File: 14-longest-common-prefix/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    let prefix = ""
    let currentLetter = ""
    let i =0;
    while(true) {
        let equal = true;
        let letter = strs[0][i];
        for (let word of strs) {
            if (i > word.length-1 || word[i] !== letter ) {
                equal = false;
                console.log(`False found for word ${word} and its letter ${word[i]}`)
                break;
            }
        }
        if (!equal) {
            break;
        }
        prefix += letter;
        i++;
    }
    return prefix
};